scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/FileSystemService.java

```diff
@@ -146,4 +145,37 @@ public void createFileInDirectory(String username, String type, String id, Strin
         }
     }
 
+    @Override
+    public void createZipFile(String username, String type, String id, String version)
+            throws FileSystemStorageException, IOException {
+
+        Path pathOfZip = createPath(username, type, id, version);
+        if (pathOfZip != null) {
+            String directoryOfZip = pathOfZip.toString();
+            String zipFileName = directoryOfZip + File.separator + type + "_" + id + "_version_" +
+                    version + ".zip";
+            FileOutputStream fileOutputStream = new FileOutputStream(zipFileName);
+            ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream);
+            File fileToZip = new File(directoryOfZip);
+
+            File[] children = fileToZip.listFiles();
+            assert children != null;
+            for (File childFile : children) {
+                if (!zipFileName.contains(childFile.getName()) && !childFile.isHidden()) {
+                    FileInputStream fileInputStream = new FileInputStream(childFile);
+                    ZipEntry zipEntry = new ZipEntry(childFile.getName());
+                    zipOutputStream.putNextEntry(zipEntry);
+
+                    byte[] bytes = new byte[4096];
+                    int length;
+                    while ((length = fileInputStream.read(bytes)) >= 0) {
+                        zipOutputStream.write(bytes, 0, length);
+                    }
+                    fileInputStream.close();
```

Comment:
Maybe use a try-with-resource statement here (https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) to make the code a little more readable?